File: common/src/main/java/com/lib/common/network/http/interceptor/HeaderInterceptor.java
package com.lib.common.network.http.interceptor;

import java.io.IOException;
import java.util.Map;

import okhttp3.Headers;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * 请求头部信息拦截器
 */

public class HeaderInterceptor implements Interceptor {
	private Map<String, String> mHeaders;

	public HeaderInterceptor(Map<String, String> header) {
		mHeaders = header;
	}

	@Override
	public Response intercept(Chain chain) throws IOException {
		Request request = chain.request().newBuilder()
				.headers(Headers.of(mHeaders))
				.build();
		return chain.proceed(request);
	}
}

File: common/src/main/java/com/lib/common/network/http/interceptor/HeaderInterceptor.kt
package com.lib.common.network.http.interceptor

import java.io.IOException

import okhttp3.Headers
import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response

/**
 * 请求头部信息拦截器
 */

class HeaderInterceptor(private val mHeaders: Map<String, String>) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request().newBuilder()
                .headers(Headers.of(mHeaders))
                .build()
        return chain.proceed(request)
    }
}
